thousandeyes/resource_alert_rule_test.go

```diff
@@ -0,0 +1,79 @@
+package thousandeyes
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccThousandEyesAlertRule(t *testing.T) {
+	var alertRuleResourceName = "thousandeyes_alert_rule.test"
+	testCases := []struct {
+		name                 string
+		createResourceFile   string
+		updateResourceFile   string
+		checkDestroyFunction func(*terraform.State) error
+		checkCreateFunc      []resource.TestCheckFunc
+		checkUpdateFunc      []resource.TestCheckFunc
+	}{
+		{
+			name:               "test_association_maintained_after_alert_rule_update",
+			createResourceFile: "acceptance_resources/alert_rule/create_test_with_alert_rule.tf",
+			updateResourceFile: "acceptance_resources/alert_rule/update_alert_rule.tf",
+			checkDestroyFunction: func(state *terraform.State) error {
+				resourceList := []ResourceType{
+					{
+						Name:         "Agent To Server Test",
+						ResourceName: "thousandeyes_agent_to_server",
+						GetResource: func(id int64) (interface{}, error) {
+							return testClient.GetAgentServer(id)
+						}},
+					{
+						Name:         "Agent To Server Alert Rule Test",
+						ResourceName: "thousandeyes_alert_rule",
+						GetResource: func(id int64) (interface{}, error) {
+							return testClient.GetAlertRule(id)
+						}},
+				}
+				return testAccCheckResourceDestroy(resourceList, state)
+			},
+			checkCreateFunc: []resource.TestCheckFunc{
+				resource.TestCheckResourceAttr(alertRuleResourceName, "rounds_violating_required", "3"),
+			},
+			checkUpdateFunc: []resource.TestCheckFunc{
+				resource.TestCheckResourceAttr(alertRuleResourceName, "rounds_violating_required", "4"),
+				resource.TestCheckResourceAttr(alertRuleResourceName, "test_ids.#", "1"),
```

Comment:
We may want to start adding some comments to these Acceptance tests - the ones that I created also do not have them and they should!
It's not immediate to understand that here we are making sure that if an `alertRule` is updated, them the `tests` associated with it are not cleared. Can you add a comment?

Reply:
tried to reflect that in the test name but i guess i failed miserably 😅 
i've added the comments, take another look when possible